src/GameBoard/GameBoardSaver.java

```diff
@@ -30,77 +45,155 @@ public static void updateGameBoard(String fileName) {
         try {
             // Load the existing XML file
             File inputFile = new File(fileName);
-            if (!inputFile.exists()) {
-                System.out.println("File not found: " + inputFile.getAbsolutePath());
-                return; // Exit if the file is not found
-            }
-
             DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
             DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
             Document doc = dBuilder.parse(inputFile);
             doc.getDocumentElement().normalize();
 
             Scanner scanner = new Scanner(System.in);
-            String continueUpdating = "yes"; // Initialize to "yes" to enter the loop
 
-            while (continueUpdating.equalsIgnoreCase("yes")) {
+            // Loop indefinitely for updates
+            while (true) {
                 System.out.println("Choose an option:");
                 System.out.println("1. Update square rent");
                 System.out.println("2. Update square position");
                 System.out.println("3. Update square name and price");
-                System.out.println("4. Exit");
+                System.out.println("4. Update square type");
+                System.out.println("5. Exit");
+

```

Comment:
1 & 3 can be combined : "Update property"
5 should return to the main instead of killing the app (just to confirm)